src/offset/library.cairo

```diff
@@ -312,307 +277,192 @@ namespace CarbonableOffseter {
     }
 
     func deposit{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        token_id: Uint256
+        token_id: Uint256, value: Uint256
     ) -> (success: felt) {
         alloc_locals;
 
-        // [Security] Start reetrancy guard
-        ReentrancyGuard.start();
-
         // [Check] Uint256 compliance
         with_attr error_message("CarbonableOffseter: token_id is not a valid Uint256") {
             uint256_check(token_id);
         }
+        with_attr error_message("CarbonableOffseter: value is not a valid Uint256") {
+            uint256_check(value);
+        }
 
-        // [Interaction] Transfer token_id from caller to contract
-        let (carbonable_project_address) = CarbonableOffseter_carbonable_project_address_.read();
-        let (caller) = get_caller_address();
-        let (contract_address) = get_contract_address();
-        IERC721.transferFrom(
-            contract_address=carbonable_project_address,
-            from_=caller,
-            to=contract_address,
-            tokenId=token_id,
-        );
-
-        // [Check] Transfer successful
-        let (owner) = IERC721.ownerOf(
-            contract_address=carbonable_project_address, tokenId=token_id
-        );
-        with_attr error_message("CarbonableOffseter: transfer failed") {
-            assert owner = contract_address;
+        // [Check] Deposited value is not null
+        let zero = Uint256(low=0, high=0);
+        let (is_zero) = uint256_eq(value, zero);
+        with_attr error_message("CarbonableOffseter: value is null") {
+            assert is_zero = FALSE;
         }
 
-        // [Effect] Register the caller with the token id and the current timestamp
-        let (current_time) = get_block_timestamp();
-        CarbonableOffseter_registered_owner_.write(token_id, caller);
-        CarbonableOffseter_registered_time_.write(token_id, current_time);
+        // [Effect] If owned token id exists then deposit to (create a new token)
+        let (to_token_id) = _get_token_id();
+        let (is_zero) = uint256_eq(to_token_id, zero);
+        if (is_zero == TRUE) {
+            let (contract_address) = get_contract_address();
+            let (success) = _deposit(
+                from_token_id=token_id, to_token_id=zero, to=contract_address, value=value
```

Comment:
What's the advantage of this approach VS creating the contract token_id at initialization?

Reply:
I don't think the contract has the rights to mint a new token id while initializing, does it?